### ISSUE TITLE
MacOSX: TextEdit event inconsistence

### ISSUE THREAD

**Issue by pablomartinezm** [gorakhargosh/watchdog#538]
Hi again,

I'm attempting to monitor some TextEdit events on MacOSX and there's an inconsistent event sequence:

1. Create a file:
```
<FileCreatedEvent: src_path='/text.txt'>
```

2. Edit + Save the file:
```
<FileMovedEvent: src_path='/text.txt', dest_path='/text.txt.sb-fc8ad3a6-66GHcO'>
<DirModifiedEvent: src_path='/'>
<FileDeletedEvent: src_path='/text.txt.sb-fc8ad3a6-66GHcO'>
```

So I end up with no file :/

any help?

**Comment by BoboTiG** [comment]
Don't you have a `FileCreatedEvent` at the end?

**Comment by SamSchott** [comment]
I have encountered a similar issue. Yes, there is an additional `FileCreatedEvent`, but for some reason it gets logged before the above three. Here is the output from the `LoggingEventHandler` while saving changes to "Untitled.txt":

```
2019-12-10 09:34:50 - Created file: /Untitled.txt
2019-12-10 09:34:50 - Moved file: from /Untitled.txt to /Untitled.txt.sb-b78ef837-68vb68
2019-12-10 09:34:50 - Modified directory: /
2019-12-10 09:34:50 - Deleted file: /Untitled.txt.sb-b78ef837-68vb68
2019-12-10 09:34:50 - Modified directory: /
```

**Comment by BoboTiG** [comment]
@SamSchott, am I right if I say that recent commits fix that issue?

**Comment by SamSchott** [comment]
Yes, this should be fixed now. I am currently working an PR to somewhat improve the handling of coalesced events, as noted in #749 there are still a large number of spurious events, but the actual order is correct now.

**Comment by BoboTiG** [comment]
Fixed with #734 and #750.
